### PR TITLE
Add docker container

### PR DESCRIPTION
**Pull request by ellisrichardj**
The docker image has all dependencies to run `btb-forestry` including:
`augur` with adaptations to add branch lengths to trees when viewed, `mega-cc`, `snp-dists` and `snp-sites`.  It is not necessary to use `conda` for `augur` installation, thereby simplifying the requirements.

This means that the process will be portable and reproducible.  Changes to the config and main script were required to enable the pipeline to run properly using the `-with-docker` switch.

### CODE REVIEW

File: docker/dockerfile [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    curl \
+    unzip \
+    gcc \
+    git \
+    build-essential \
+    python3-dev \
+    zlib1g-dev \
+    libsqlite3-dev \
+    python3-pip \
+    snp-sites
```
Comment (nick-pestell) [inline docker/dockerfile:22]: Are all of these definitely needed? E.g. `build-essential`, `python3-dev`, `zlib1g-dev` and `libsqlite3-dev`? I might be wrong but I think maybe there not required for any of the packages you install ...

**Review by nick-pestell — CHANGES_REQUESTED** [review]
Great to have this in. A few things:
1. see comments in the files;
1. rename `dockerfile` to `Dockerfile`;
2. Maybe it's unnecessary to have the `docker` directory. Could just have `Dockerfile` in the base...
3. Is pandas installed anywhere? Couldn't see it.

File: nextflow.config [hunk 2]
```diff
@@ -10,15 +10,19 @@ manifest {
...
+}
+*/
+// docker configuration
 docker { 
     enabled = true
     fixOwnership = true
-    runOptions = '-v ~:/home' }
-*/
+    runOptions = '--user "$(id -u):$(id -g)"'
+    conda.enabled = false
```
[the remark this answers is not in the document]
Comment (ellisrichardj) [reply nextflow.config:23]: I was thinking of retaining the option to use conda if run locally, so have kept the commented lines for now.

File: docker/dockerfile (hunk 1 above)
Comment (nick-pestell) [reply docker/dockerfile:22]: Ok, cool. I think you can get `make` on it's own without the extra bits in `build-essential`

Comment (ellisrichardj) [reply docker/dockerfile:22]: Ok - will swap `build-essential` for `make`

**Comment by ellisrichardj** [comment]
Comments addressed and Dockerfile moved and renamed.  `Pandas` (and plenty of other python libraries are installed as part of the `Augur` package, so no need to install separately.  Will hopefully avoid any conflicts too

**Review by nick-pestell — APPROVED** [review]
Thanks for making those changes

**Pull request merged**